src/ui/components/results/verticalresultscomponent.js

```diff
@@ -152,11 +163,21 @@ export default class VerticalResultsComponent extends Component {
     this.results = [];
     this.numColumns = this._config.maxNumberOfColumns;
 
+    const appliedFiltersOpts = this._config.appliedFilters || {};
+
     /**
      * Config options used in the {@link ResultsHeaderComponent}
      */
-    const { showFieldNames, resultsCountSeparator, showResultCount, showAppliedFilters, showChangeFilters } = this._config;
-    this.resultsHeaderOpts = { showFieldNames, resultsCountSeparator, showResultCount, showAppliedFilters, showChangeFilters };
+    this.resultsHeaderOpts = {
```

Comment:
In this component and any others, I want us to avoid referring to deprecated configuration options. It adds a lot of complexity to a component. We should use the VerticalResultsConfig as a translation mechanism. It takes in user-supplied configuration (potentially with deprecated options) and converts it to an object that the VerticalResultsComponent can understand. In this case, the VerticalResults component should now only refer to the appliedFilters object. It should not refer to things like _config.showFieldNames any more. 

The VerticalResultsConfig class has the responsibility of converting 'showFieldNames' into 'appliedFilters.showFieldNames', etc. For those configuration options that were removed and renamed, we should have their default values be the values of the legacy attributes. So, appliedFilters.showFieldNames would default to showFieldNames, which defaults to false. VerticalResultsConfig should produce an up-to-date, correct configuration for VerticalResults. That means a fully populated appliedFilters object with any defaults. 

I'd like to avoid using the "parseConfig" method as well. I think such a class could definitely be useful, but it warrants a bit more discussion.

Reply:
updated to only use new config in vertical results (outside of the config class).